### PR TITLE
Eng 1595 write dev documentation for complex sql

### PR DESCRIPTION
**Pull request by eunice-chan**
## Describe your changes and why you are making these changes
Commented on all complex SQL queries in `src/golang/lib/collections` and `src/golang/cmd/server/queries` to make it easier to understand them without reading too closely.

## Related issue number (if any)
Eng 1595

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [N/A] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [N/A] If this is a new feature, I have added unit tests and integration tests.
- [N/A] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [N/A] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/cmd/server/queries/postgres.go [hunk 1]
```diff
@@ -83,6 +87,10 @@ func (r *postgresReaderImpl) GetLoadOperatorSpecByOrganization(
 	organizationId string,
 	db database.Database,
 ) ([]LoadOperatorSpecResponse, error) {
+	// Get the artifact id, artifact name, operator id, workflow name, workflow id,
+	// and operator spec of all load operators (to_id) and the artifact(s) going to
+	// that operator (from_id; the table/object being written/saved) in the workflows
```
Comment (saurav-c) [inline src/golang/cmd/server/queries/postgres.go:92]: What does "the table/object being written/saved" referred to? Is that part of the operator spec being returned?

Comment (saurav-c) [inline src/golang/cmd/server/queries/postgres.go:92]: What does "the table/object being written/saved" referred to? Is that part of the operator spec being returned?

File: src/golang/cmd/server/queries/standard.go [hunk 2]
```diff
@@ -40,6 +41,8 @@ func (r *standardReaderImpl) GetArtifactIdsFromWorkflowDagIdsAndDownstreamOperat
 		return nil, errors.New("Provided empty IDs list.")
 	}
 
+	// Get all the unique `artifact_id`s with an outgoing edge to an operator with id `operatorIds`
```
Comment (saurav-c) [inline src/golang/cmd/server/queries/standard.go:44]: nit: id in `operatorIds`

**Review by saurav-c — APPROVED** [review]
This is great! Just left a couple of nits.

**Review by likawind — APPROVED** [review]
Yay!

File: src/golang/cmd/server/queries/postgres.go (hunk 1 above)
Comment (eunice-chan) [reply src/golang/cmd/server/queries/postgres.go:92]: Updated to:
```
	// Get the artifact id, artifact name, operator id, workflow name, workflow id,
	// and operator spec of all load operators (`to_id`s) and the artifact(s) going to
	// that operator (`from_id`s; these artifacts are the objects that will be saved 
	// by the operator to the integration) in the workflows owned by the specified
	// organization.
```

File: src/golang/cmd/server/queries/standard.go (hunk 2 above)
Comment (eunice-chan) [reply src/golang/cmd/server/queries/standard.go:44]: Updated to:
```
	// Get all the unique `artifact_id`s with an outgoing edge to an operator specified by `operatorIds`
	// from workflow DAGs specified by `workflowDagIds`.
```

**Pull request merged**